apps/web/pages/api/book/event.ts

```diff
@@ -231,6 +232,11 @@ async function ensureAvailableUsers(
   }
   return availableUsers;
 }
+const checkIfValueIsPresent = (val: string | boolean) => {
+  if (typeof val == "boolean" && val === false) return false;
```

Comment:
A boolean can have false as a valid value, I don't think false should be considered empty. `undefined` or `null` is empty. Due to this, a checkbox if marked required isn;t allowing any other value value except true.

Reply:
i thought if a checkbox is marked required that it should be true (like i agree terms and conditions ). there is only one checkbox so how do we differentiate whether user wanted to mark false or just skip it and mark empty ?

Reply:
Alternative approach:
```javascript
const checkIfValueIsPresent = (val: string | boolean) => {
  if (typeof val == "string" && val.trim() === "") return false;
  return !!val;
};

Reply:
@emrysal ok we can use this but i am trying out zod here like

`export const checkStringOrBooleanExist = z.union([z.string().min(1), z.boolean()]);`

```
  if (eventType.customInputs) {
    eventType.customInputs.forEach((customInput) => {
      if (
        customInput.required === true &&
        (reqBody.customInputs.length === 0 ||
          reqBody.customInputs.filter(
            ({ label, value }) => label === customInput.label && checkStringOrBooleanExist.parse(value)
          ).length != 1)
      ) {
        throw new Error("Missing required input");
      }
    });
  }
```
which one should i use ?

and  how would required custom input checkbox would work ? do we allow not marking it even if it is a required input . well in this case both required and not required would work the same (you can mark it or leave it).

Reply:
What do you think about the version below?

```typescript
const requiredCustomInputSchema = z.union([
  // string must be given & nonempty
  z.string().trim().min(1),
  // boolean must be true if set.
  z.boolean().refine((v) => v === true),
]);

if (eventType.customInputs) {
    eventType.customInputs.forEach((customInput) => {
      if (customInput.required === true) {
        requiredCustomInputSchema.parse(reqBody.customInputs.find(
          userInput => userInput.label === customInput.label)?.value
        ));
      }
    });
  }
```

`.parse` throws an error of its own, so you can use that instead of throwing a custom error. Alternatively use `.safeParse` and read the .success return value.

Reply:
This one is much better 👍

Reply:
done